v-next/hardhat-keystore/src/index.ts

```diff
@@ -0,0 +1,70 @@
+import type { HardhatPlugin } from "@ignored/hardhat-vnext/types/plugins";
+
+import { ArgumentType, task } from "@ignored/hardhat-vnext/config";
+
+import { get, list, remove, set } from "./methods.js";
+
+export const PLUGIN_ID = "hardhat-keystore";
+
+const hardhatKeystorePlugin: HardhatPlugin = {
+  id: PLUGIN_ID,
+  hookHandlers: {
+    configurationVariables: import.meta.resolve(
+      "./hook-handlers/configuration-variables.js",
+    ),
+  },
+  tasks: [
+    task("keystore", "Store your keys in a secure way")
+      .setAction(async (_, _hre) => {})
+      .build(),
+
+    task(
+      ["keystore", "set"],
+      "Sets a new value in the keystore associated with the specified key",
+    )
+      .addPositionalArgument({
+        name: "key",
+        type: ArgumentType.STRING,
+        description: "Specifies the key to set in the keystore",
+      })
+      .addFlag({
+        name: "force",
+        description: "Forces overwrite if the key already exists.",
+      })
+      .setAction(async ({ key, force }) => {
+        // await set(key, hre);
+        await set(key, force);
+      })
+      .build(),
+
+    task(["keystore", "get"], "Get a value given a key")
+      .addPositionalArgument({
+        name: "key",
+        type: ArgumentType.STRING,
+        description: "Specifies the key to retrieve the value for",
+      })
+      .setAction(async ({ key }) => {
+        await get(key);
```

Comment:
Should this be returning a the value, so it can be used programmatically in a Hardhat script?

Reply:
I think that the keystore should not be used programmatically considering its purpose. For example we do not want people to use it in CI because how would they input the password? 
@alcuadrado